packages/berry-fslib/sources/NodeFS.ts

```diff
@@ -194,13 +194,29 @@ export class NodeFS extends FakeFS {
 
     return await new Promise<void>((resolve, reject) => {
       this.realFs.symlink(NodeFS.fromPortablePath(target.replace(/\/+$/, ``)), NodeFS.fromPortablePath(p), type, this.makeCallback(resolve, reject));
+    }).catch((e: any) => {
+      // Pre Win 10 support
+      if (process.platform === `win32` && e.code === 'EPERM') {
+        this.writeFilePromise(p, target.replace(/[\\\/]$/, ''));
+      } else {
+        throw e;
+      }
     });
   }
 
   symlinkSync(target: string, p: string) {
     const type: 'dir' | 'file' = target.endsWith(`/`) ? `dir` : `file`;
 
-    return this.realFs.symlinkSync(NodeFS.fromPortablePath(target.replace(/\/+$/, ``)), NodeFS.fromPortablePath(p), type);
+    try {
+      return this.realFs.symlinkSync(NodeFS.fromPortablePath(target.replace(/\/+$/, ``)), NodeFS.fromPortablePath(p), type);
+    } catch (e) {
+      // Pre Win 10 support
+      if (process.platform === `win32` && e.code === 'EPERM') {
```

Comment:
Can you abstract this in a `isLegacyWindows()`? I'd also prefer to add a check to disable this codepath if the first part of `os.release()` is 10 or higher.